### PR TITLE
Add ci-kubernetes-e2e-gce-min-node-permissions

### PR DESCRIPTION
**Pull request by ihmccreery**
... to test granting minimal node permissions

Xref https://github.com/kubernetes/kubernetes/pull/52868

### CODE REVIEW

File: jobs/config.json [hunk 1]
```diff
@@ -3699,6 +3699,23 @@
       "generated"
     ]
   },
+  "ci-kubernetes-e2e-gce-min-node-permissions": {
+    "args": [
+      "--check-leaked-resources",
+      "--env-file=jobs/platform/gce.env",
+      "--env-file=jobs/env/ci-kubernetes-e2e-gce-min-node-permissions.env",
+      "--extract=ci/latest",
+      "--gcp-project=k8s-min-node-permissions",
```
Comment (krzyzacy) [inline jobs/config.json]: does this project has some special quota/IAM?

File: jobs/config.json [hunk 2]
```diff
@@ -3699,6 +3699,23 @@
...
+      "--gcp-zone=us-central1-f",
+      "--provider=gce",
+      "--test_args=--ginkgo.skip=\\[Flaky\\]|\\[Feature:.+\\] --minStartupPods=8",
+      "--timeout=500m"
+    ],
+    "scenario": "kubernetes_e2e",
+    "sigOwners": [
+      "UNKNOWN"
```
Comment (krzyzacy) [inline jobs/config.json]: put a sig here

File: jobs/config.json (hunk 1 above)
Comment (ihmccreery) [reply jobs/config.json]: It has the min-node-permissions@k8s-min-node-permissions.iam.gserviceaccount.com service account configured.

File: jobs/config.json (hunk 2 above)
Comment (ihmccreery) [reply jobs/config.json]: Done.

**Comment by ihmccreery** [comment]
Ptal.

File: testgrid/config/config.yaml [hunk 3]
```diff
@@ -1995,6 +1995,8 @@ dashboards:
     test_group_name: ci-kubernetes-e2e-gce-device-plugin-gpu
   - name: gce-gpu
     test_group_name: ci-kubernetes-e2e-gce-gpu
+  - name: gce-min-node-permissions
+    test_group_name: ci-kubernetes-e2e-gce-min-node-permissions
```
Comment (krzyzacy) [inline testgrid/config/config.yaml]: you'll need to add something similar to https://github.com/ihmccreery/test-infra/blob/21a4014b4bd783ce58a7c93dee03a42624692bf6/testgrid/config/config.yaml#L1735-L1736 for the new job, and you should be good to go.

**Comment by krzyzacy** [comment]
(and please run `bazel run //jobs:config_sort` to keep the config sorted)

File: testgrid/config/config.yaml (hunk 3 above)
Comment (ihmccreery) [reply testgrid/config/config.yaml]: Done.

**Comment by ihmccreery** [comment]
> (and please run bazel run //jobs:config_sort to keep the config sorted)

Done.

File: testgrid/config/config.yaml [hunk 4]
```diff
@@ -1758,6 +1758,8 @@ test_groups:
   gcs_prefix: kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-gci-stable1-gci-stable2-downgrade-cluster
 - name: ci-kubernetes-e2e-gke-gci-stable1-gci-stable2-downgrade-cluster-parallel
   gcs_prefix: kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-gci-stable1-gci-stable2-downgrade-cluster-parallel
+- name: gce-min-node-permissions
+  gcs_prefix: kubernetes-jenkins/logs/gce-min-node-permissions
```
Comment (krzyzacy) [inline testgrid/config/config.yaml]: s/gce-min-node-permissions/ci-kubernetes-e2e-gce-min-node-permissions/

Comment (ihmccreery) [reply testgrid/config/config.yaml]: Done.

**Comment by krzyzacy** [comment]
(seems prow/config.yaml is still not sorted properly - LGTM once all presubmit passes)

**Comment by ihmccreery** [comment]
*sigh* Okay @krzyzacy ptal.

**Comment by ihmccreery** [comment]
Thanks @krzyzacy.  When can I expect this to show up on the dashboards?

**Comment by krzyzacy** [comment]
@ihmccreery http://k8s-testgrid.appspot.com/google-gce#gce-min-node-permissions

**Pull request merged**